src/misc/cross.cpp

```diff
@@ -392,35 +393,34 @@ FILE *fopen_wrap(const char *path, const char *mode) {
 			if ( ( strlen(check) == 5 && strcmp(check,"/proc") == 0) || strncmp(check,"/proc/",6) == 0) {
 //				LOG_MSG("lst hit %s blocking!",path);
 				free(check);
-				return nullptr;
-			}
-			free(check);
-		}
-	}
-
-#if 0
-//Lightweight version, but then existing files can still be read, which is not ideal	
-	if (strpbrk(mode,"aw+") != NULL) {
-		LOG_MSG("pbrk ok");
-		char* check = realpath(path,NULL);
-		//Will be null if file doesn't exist.... ENOENT
-		//TODO What about unlink /proc/self/mem and then create it ?
-		//Should be safe for what we want..
-		if (check) {
-			if (strncmp(check,"/proc/",6) == 0) {
-				free(check);
-				return NULL;
+				return false;
 			}
 			free(check);
 		}
 	}
-*/
-#endif //0 
 
 #endif //HAVE_REALPATH
 #endif
+	return true;
+}
+
+bool is_path_allowed(const std_fs::path& path)
+{
+#if defined(WIN32)
+	// path.c_str() would be UTF-16 encoded
+	return is_path_allowed(path.string().c_str());
+#else
+	return is_path_allowed(path.c_str());
+#endif
```

Comment:
I'd probably get rid of the `#ifdef` and just do `path.string().c_str()` everywhere.  That's what I've done elsewhere in the code in any case.

Reply:
This would have triggered static analysis warning about unnecessary conversion. That's why you can find `path.string().c_str()` in f`s_util_win32.cpp`, but `path.c_str()` in `fs_util_posix.cpp`